### ISSUE TITLE
Blackberry OS 10.3 can't stop background mobile data usage

### ISSUE THREAD

**Issue by scienceopen** [schildbach/bitcoin-wallet#263]
The app consumes about 30-40MByte of data per day, even if not being used. Installed from the Blackberry App World, version 1.0.225.0. 
Even if the phone is rebooted and Bitcoin isn't opened, Bitcoin app will keep using 30MB or so per day.
Even when Run When Minimized is disabled, the data use continues.

Maybe an option could be made within the app to only use WiFi network when in the background.

I understand a contributor to this issue could be a defect in Android Player on Blackberry OS 10.3.2, since it shouldn't allow Run When Minimized apps to keep running, especially when the app is closed! Maybe Bitcoin is running in a headless mode on Blackberry?

Thank you for taking the time to make the Android wrapped Blackberry version.

**Comment by schildbach** [comment]
Can you attach your app logfiles of the last days?

**Comment by scienceopen** [comment]
From within the Bitcoin app itself? I'm a little reluctant as it seems to want to send the log directly from the app, or does it save the log to a file that I manually send? I did a complete uninstall/reinstall, rebooting phone and then in a couple days about 1.3GByte was used by the app. I decided to just install the app for now. 

I understand the need to keep synced with everyone else, but it seems like the app shouldn't be downloading GBytes of data when it's not even open, and more importantly it's not respecting the "do not run when minimized" option. Is the app also acting this way on native Android?  It is downloading far more data than any other app I have, including Youtube.

**Comment by schildbach** [comment]
It hands the logfile over to your email client, which will attach it and send it to our mailbox. If you prefer, you can replace the email address with your own and have a look at it first.

The app needs to run in background in order to notify the user about incoming payments. I'm not sure what you mean with "minimized".

**Comment by scienceopen** [comment]
OK now I understand better that the app needs to run what in Blackberry parlance is called headless mode. So that is expected behavior. I suppose this would turn into a *feature request* to allow non-heavy users to select to update the Bitcoin stream **only when on WiFi**

**Comment by schildbach** [comment]
On Android, this feature is built into the OS. You can forbid individual apps to use metered connections in the background. Therefore I'm not really keen on duplicating this work.

**Comment by scienceopen** [comment]
Yes it seems that the Android runtime in the bby os isn't respecting this setting for this app. I know it works for non-Android ported apps e.g. native, cascades, etc. where I can stop background data on bby os 10.

Given the bby priv (android os) is shipping this Friday, the small slice of users using this Bitcoin app from Blackberry App World on bby os 10 will shrink even further, so perhaps WONTFIX is appropriate here.

**Comment by scienceopen** [comment]
It may however be appropriate to put a note in the blackberry app world description that background data will be used, even on cellular, or someone could be very upset if they weren't on an unlimited data plan. It could actually use their monthly data quota in a couple days.
